ui: clusterviz: statsview refactor

vilterp opened the pull request:
Break StatsView down into subcomponents which are used from NodeView and LocalityView, so that those two can handle elements which are different between them, like the liveness indicator and node vs locality icon.

Release note: None

Commit:
ui: clusterviz: refactor NodeView/LocalityView/StatsView

Break StatsView down into subcomponents which are used from NodeView and
LocalityView, so that those two can handle elements which are different
between them, like the liveness indicator and node vs locality icon.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/22671)
<!-- Reviewable:end -->

couchand commented:
I was going to try it out first, but am looking at other things at the moment.  If it blends, LGTM

---

Review status: 0 of 7 files reviewed at latest revision, 5 unresolved discussions, all commit checks successful.

---

*[pkg/ui/ccl/src/views/clusterviz/components/nodeOrLocality/capacityArc.tsx, line 10 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/22671#-L5K7PT0Snk7sZyX0YMz:-L5K7PT1_B5-yAoYlw_n:b-itmgp7) ([raw file](https://github.com/cockroachdb/cockroach/blob/1a0ece2be2652c322aa2fdc80970f3c9911e12a3/pkg/ui/ccl/src/views/clusterviz/components/nodeOrLocality/capacityArc.tsx#L10)):*
> ```XML
> 
> import React from "react";
> import * as PathMath from "ccl/src/views/clusterviz/containers/map/pathmath";
> ```

we should probably move that pathmath file somewhere else (`clusterviz/util` maybe?)

also, no "ccl" here and elsewhere.  though the fact that this keeps coming up makes me wonder if we should do one of two things, either:

- rethink the build/import strategy to account for your tooling, or
- not worry about it for now and make a pass over everything later to clean it up.

---

*[pkg/ui/ccl/src/views/clusterviz/components/nodeOrLocality/capacityArc.tsx, line 38 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/22671#-L5K8PnOIYbpcijLWUBt:-L5K8GyM_TPXayv5sIe9:b-5jx138) ([raw file](https://github.com/cockroachdb/cockroach/blob/1a0ece2be2652c322aa2fdc80970f3c9911e12a3/pkg/ui/ccl/src/views/clusterviz/components/nodeOrLocality/capacityArc.tsx#L38)):*
> ```XML
>     return (
>       <g>
>         <g transform="translate(90 115)">
> ```

what are these magic numbers?  also the ones on the `text`s below...  maybe these would move into a stylesheet in a future pass

---

*[pkg/ui/ccl/src/views/clusterviz/components/nodeOrLocality/labels.tsx, line 10 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/22671#-L5K8o5F1inhsIMinCwI:-L5K8o5GO54Lm11G7kCa:bsv6wc0) ([raw file](https://github.com/cockroachdb/cockroach/blob/1a0ece2be2652c322aa2fdc80970f3c9911e12a3/pkg/ui/ccl/src/views/clusterviz/components/nodeOrLocality/labels.tsx#L10)):*
> ```XML
> 
> import React from "react";
> import { DARK_BLUE, MAIN_BLUE } from "oss/ccl/src/views/shared/colors";
> ```

`oss/ccl`???

---

*[pkg/ui/ccl/src/views/clusterviz/components/nodeOrLocality/labels.tsx, line 26 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/22671#-L5K944RaFsEKpxmA2Zy:-L5K944RaFsEKpxmA2Zz:b-z1q6p7) ([raw file](https://github.com/cockroachdb/cockroach/blob/1a0ece2be2652c322aa2fdc80970f3c9911e12a3/pkg/ui/ccl/src/views/clusterviz/components/nodeOrLocality/labels.tsx#L26)):*
> ```XML
>         fontWeight="bold"
>         letterSpacing="1"
>         transform="translate(50 22)"
> ```

This seems like a strange place to do the translation.  I tend to expect a component to render itself wherever it is, not at a fixed offset from there.  I suppose we just aren't thinking of these as reusable components (which raises the question of if they should stand on their own like this...)

---

*[pkg/ui/ccl/src/views/clusterviz/containers/map/localityView.tsx, line 53 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/22671#-L5K9i0ENO47gUENtmHx:-L5K9i0ENO47gUENtmHy:b7ik4b2) ([raw file](https://github.com/cockroachdb/cockroach/blob/1a0ece2be2652c322aa2fdc80970f3c9911e12a3/pkg/ui/ccl/src/views/clusterviz/containers/map/localityView.tsx#L53)):*
> ```XML
>         onClick={this.onClick}
>         style={{ cursor: "pointer" }}
>         fill="none"
> ```

did you figure out why this fill is needed?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/22671#-:-L5KCMam5vnhfuF17miV:b-wog4og)*
<!-- Sent from Reviewable.io -->

vilterp commented:
Review status: 0 of 9 files reviewed at latest revision, 5 unresolved discussions, some commit checks pending.

---

*[pkg/ui/ccl/src/views/clusterviz/components/nodeOrLocality/capacityArc.tsx, line 10 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/22671#-L5K7PT0Snk7sZyX0YMz:-L5KJ99JfKsEWGIbYuod:b-lwtd8r) ([raw file](https://github.com/cockroachdb/cockroach/blob/1a0ece2be2652c322aa2fdc80970f3c9911e12a3/pkg/ui/ccl/src/views/clusterviz/components/nodeOrLocality/capacityArc.tsx#L10)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

we should probably move that pathmath file somewhere else (`clusterviz/util` maybe?)

also, no "ccl" here and elsewhere.  though the fact that this keeps coming up makes me wonder if we should do one of two things, either:

- rethink the build/import strategy to account for your tooling, or
- not worry about it for now and make a pass over everything later to clean it up.
</blockquote></details>

I actually misremembered the rule and thought that CCL code was supposed to be imported with the `ccl` prefix. Cleaned it up. Having something error out if you're doing it wrong would be nice (simple linter?) Not sure the best way to implement this.

---

*[pkg/ui/ccl/src/views/clusterviz/components/nodeOrLocality/capacityArc.tsx, line 38 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/22671#-L5K8PnOIYbpcijLWUBt:-L5KEF5oP0Lk4_p4d1eq:b-gepkcj) ([raw file](https://github.com/cockroachdb/cockroach/blob/1a0ece2be2652c322aa2fdc80970f3c9911e12a3/pkg/ui/ccl/src/views/clusterviz/components/nodeOrLocality/capacityArc.tsx#L38)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

what are these magic numbers?  also the ones on the `text`s below...  maybe these would move into a stylesheet in a future pass
</blockquote></details>

Make the thing centered instead of from the top left. Probably adding the clickable background rect would be a good time to rationalize this.

---

*[pkg/ui/ccl/src/views/clusterviz/components/nodeOrLocality/labels.tsx, line 10 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/22671#-L5K8o5F1inhsIMinCwI:-L5KDjqx8CpzfT6JJjGg:bgax84s) ([raw file](https://github.com/cockroachdb/cockroach/blob/1a0ece2be2652c322aa2fdc80970f3c9911e12a3/pkg/ui/ccl/src/views/clusterviz/components/nodeOrLocality/labels.tsx#L10)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

`oss/ccl`???
</blockquote></details>

Yeah, missed this one. Amazing that that's what intellij comes up with, and that it works.

---

*[pkg/ui/ccl/src/views/clusterviz/components/nodeOrLocality/labels.tsx, line 26 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/22671#-L5K944RaFsEKpxmA2Zy:-L5KE1_OKGa_drKPBfyS:b-4fzul3) ([raw file](https://github.com/cockroachdb/cockroach/blob/1a0ece2be2652c322aa2fdc80970f3c9911e12a3/pkg/ui/ccl/src/views/clusterviz/components/nodeOrLocality/labels.tsx#L26)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

This seems like a strange place to do the translation.  I tend to expect a component to render itself wherever it is, not at a fixed offset from there.  I suppose we just aren't thinking of these as reusable components (which raises the question of if they should stand on their own like this...)
</blockquote></details>

Good point, although doesn't seem worth the refactor right now. Putting it here does have the benefit of reducing code duplication between NodeView and LocalityView. If we wanted it to be in different places for those two, then we'd do this refactor.

---

*[pkg/ui/ccl/src/views/clusterviz/containers/map/localityView.tsx, line 53 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/22671#-L5K9i0ENO47gUENtmHx:-L5KDqh_KE2yI-l1kwt7:bb3dzq5) ([raw file](https://github.com/cockroachdb/cockroach/blob/1a0ece2be2652c322aa2fdc80970f3c9911e12a3/pkg/ui/ccl/src/views/clusterviz/containers/map/localityView.tsx#L53)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

did you figure out why this fill is needed?
</blockquote></details>

Used to change the way the sparklines rendered. No longer the case after your change. Removed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/22671)*
<!-- Sent from Reviewable.io -->
